tools/lsvast/lsvast.cpp

```diff
@@ -114,170 +179,346 @@ read_flatbuffer_file(vast::path path) {
   return result_t(ptr, flatbuffer_deleter<T>(std::move(bytes)));
 }
 
+struct indentation {
+public:
+  static constexpr const int TAB_WIDTH = 2;
+  indentation() {
+  }
+  void increase(int level = TAB_WIDTH) {
+    levels_.push_back(level);
+  }
+  void decrease() {
+    levels_.pop_back();
+  }
+
+  friend std::ostream& operator<<(std::ostream&, const indentation&);
+
+private:
+  std::vector<int> levels_;
+};
+
+class indented_scope {
+public:
+  indented_scope(indentation& indent) : indent_(indent) {
+    indent.increase();
+  }
+  ~indented_scope() {
+    indent_.decrease();
+  }
+
+private:
+  indentation& indent_;
+};
+
+std::ostream& operator<<(std::ostream& str, const indentation& indent) {
+  for (auto level : indent.levels_)
+    for (int i = 0; i < level; ++i)
+      str << ' ';
+  return str;
+}
+
 std::ostream& operator<<(std::ostream& out, const vast::fbs::uuid::v0* uuid) {
   if (!uuid || !uuid->data())
     return out << "(null)";
+  auto old_flags = out.flags();
   for (size_t i = 0; i < uuid->data()->size(); ++i)
     out << std::hex << +uuid->data()->Get(i);
+  out.flags(old_flags); // `std::hex` is sticky.
   return out;
 }
 
-void print_unknown(vast::path path, int indent) {
-  std::string spaces(indent, ' ');
-  std::cout << spaces << "(unknown " << path.str() << ")\n";
+void print_unknown(vast::path path, indentation& indent,
+                   const formatting_options&) {
+  std::cout << indent << "(unknown " << path.str() << ")\n";
 }
 
-void print_vast_db(vast::path vast_db, int indent) {
+std::string print_bytesize(size_t bytes, const formatting_options& formatting) {
+  const char* suffixes[] = {
+    " byte", " KiB", " MiB", " GiB", " TiB", " EiB",
```

Comment:
```suggestion
    " B", " KiB", " MiB", " GiB", " TiB", " EiB",
```

I just noticed in the output that this is the odd one out. I imagine you wrote it out because it could be mistaken for "billion"? If so, I'd say that's not an issue because all values have a `B` suffix.

Reply:
Actually more due to the confusion potential with "bit", but I don't mind changing it to `B`.